Src/PerformanceCollector/Shared/Implementation/IPerformanceCollector.cs

```diff
@@ -31,5 +30,12 @@ internal interface IPerformanceCollector
         /// <param name="error">Captures the error logged.</param>
         /// <param name="blockCounterWithInstancePlaceHolder">Boolean that controls the registry of the counter based on the availability of instance place holder.</param>
         void RegisterCounter(string perfCounter, string reportAs, bool isCustomCounter, out string error, bool blockCounterWithInstancePlaceHolder);
+
+        /// <summary>
+        /// Removes a counter.
+        /// </summary>
+        /// <param name="perfCounter">Name of the performance counter to remove.</param>
+        /// <param name="reportAs">ReportAs value of the counter to remove.</param>
+        void RemoveCounter(string perfCounter, string reportAs);
```

Comment:
Especially since we have favorites, chances are that if you have 10 UXs open - most of the metrics and document streams they query for will actually be exactly the same. The reason SDK needs to merge configuration from all clients is because we can't afford to apply the same filter 10 times instead of 1, send the same document 10 times instead of 1, and (most importantly for this particular case) collect the same counter 10 times instead of 1.
So we merge :)